### PR TITLE
Add front-end conditionning for closing off creation and edition of p…

### PR DESCRIPTION
**Pull request by Wardormeur**
…rojects
To be fair, a third state could be added, between open and frozen, which is basically the negation of both (registration is closed but edition is available)
This is now emulated by the negation of both because i'm fairly lazy, but I'll do it on request @DanielBrierton

### CODE REVIEW

File: client/src/event/Auth.vue [hunk 1]
```diff
@@ -7,29 +7,33 @@
...
-          <label>This form is to register or edit a project for Coolest Projects International 2018 which will be held on Saturday 26th May 2018 in RDS Simmonscourt, Dublin, Ireland. Just one person should register for each project.</label>
+          <label v-if="isOpen">This form is to register or edit a project for Coolest Projects International 2018 which will be held on Saturday 26th May 2018 in RDS Simmonscourt, Dublin, Ireland. Just one person should register for each project.</label>
+          <label v-if="!isOpen && !isFrozen">You can now only edit already registered projects for {{ event.name }}. Enter your email and you will get a link to edit your project.</label>
+          <label v-else-if="isFrozen">The registration for {{ event.name }} has now closed. Please contact {{ event.email }} if you have any questions.</label>
         </div>
       </div>
-      <div class="row">
+      <div class="row" v-if="!isFrozen">
         <div class="col">
           <input type="email" v-model="email" class="full-width-block" placeholder="you@email.com" required />
```
Comment (DanielBrierton) [inline client/src/event/Auth.vue]: Without backend checks, this will bring the user to create a new project if they haven't already got one.

**Comment by Wardormeur** [comment]
Added backend changes as well as a message upon user creation in a closed registration

File: client/src/event/Auth.vue [hunk 2]
```diff
@@ -7,29 +7,33 @@
...
-            :class="{ error: errors.has('approval') }"
-            name="approval"/>
-        </div>
-        <div class="col-6fr">
-          <p for="approval">By selecting this you are agreeing that you are 13 or over, and are happy to be contacted about this project, Coolest Projects information and related news.</p>
-          <span class="error-message" v-show="errors.has('approval:required')">* We need your permission to contact you. Please accept the terms and conditions.</span>
-        </div>
-      </label>
-      <div class="row row-double-margin">
+      <div v-if="isOpen">
```
Comment (DanielBrierton) [inline client/src/event/Auth.vue]: I feel like if we're registering users anyway, then we need to show this checkbox while `!isOpen && !isFrozen`.

Comment (DanielBrierton) [reply client/src/event/Auth.vue]: @Wardormeur Just this comment outstanding for me, otherwise 👍

**Pull request merged**